gridcomps/History/Sampler/MAPL_TrajectoryMod_smod.F90

```diff
@@ -827,8 +871,53 @@
                     'epoch_index(1:2), nx', this%epoch_index(1), &
                     this%epoch_index(2), this%nobs_epoch)
                do k=1, this%nobs_type
-                  call lgr%debug('%a %i4 %a %i12', &
-                       'obs(', k, ')%nobs_epoch', this%obs(k)%nobs_epoch )
+                  call lgr%debug('%a %i4 %a %a80 %i12 %i12 %i12', &
+                       'obs(', k, ') ', trim(this%obs(k)%name)//' %nobs_epoch, %count_location_in_matching_file, diff', &
+                       this%obs(k)%nobs_epoch, this%obs(k)%count_location_in_matching_file, &
+                       this%obs(k)%nobs_epoch - this%obs(k)%count_location_in_matching_file )
+                  !
+                  !    diagnostic print
+                  !    now we know
+                  !    ioda file split [1/2 15Z  :  1/2 21Z ]  [ 1/2 21Z :  1/2 3Z] 
+                  !        ___x  x  x x x ___ ---------------------------------- o --o ---o -- o --
+                  !           negative index (extra) at Tmin                      missing Tmax
+                  !
+                  !    count non-positive index on the left Tmin  (extra)
+                  !
+                  jj = 0
+                  do j = 1, this%obs(k)%nobs_epoch
+                     if ( this%obs(k)%location_index_ioda(j) <= 0 ) then
+                        jj = jj + 1
+                     end if
+                  end do
+
+                  kk = this%obs(k)%count_location_in_matching_file
+                  allocate ( marker(kk), _STAT )
+                  marker(:) = 0 
+                  do j = 1, this%obs(k)%nobs_epoch
+                     kk = this%obs(k)%location_index_ioda(j)
+                     if ( kk > 0 .and. kk <= this%obs(k)%nobs_epoch ) then
+                        marker(kk) = 1  ! exist
+                     end if
+                  end do
+                  do kk = 1, this%obs(k)%count_location_in_matching_file
+                     if ( marker(kk) == 0 ) then
+                        write(6, '(2x,a,2x,10i12)')  trim(this%obs(k)%name)//' missing pts in obervation ', kk
+                     end if
+                  end do
+                  deallocate(marker)
+
+                  j = this%obs(k)%nobs_epoch - this%obs(k)%count_location_in_matching_file
+                  if (j/=0) &
+                     write(6, *) trim(this%obs(k)%name)//'count_location_in_matching_file diff from cutted nobs_epoch'
+                  j = minval(this%obs(k)%location_index_ioda(:))
+                  if (j/=1) &
+                       write(6,*) trim(this%obs(k)%name)//' min =', j, &
+                       ' this%obs(k)%location_index_ioda(:) start diff from 1'
+                  j = maxval(this%obs(k)%location_index_ioda(:))
+                  if (j/=this%obs(k)%count_location_in_matching_file) &
+                       write(6,*) trim(this%obs(k)%name)//' max =', j, &
+                       ' this%obs(k)%location_index_ioda(:) end diff from obs file max pts'
```

Comment:
I realize that I should change this as debug for the NWP IODA file only, in order to prevent the general trajectory sampler case from printing out the information.

Reply:
ok